src/test/gitSecrets.test.ts

```diff
@@ -0,0 +1,108 @@
+    let accessKeyFilePath: string
+    let toolkitProjectDir: string
+    let testFixturesPath: string
+    let gitSecrets: string // path to the git-secrets executable
+
+    function setupGitSecretsExecutable(gitSecretsExecutablePath: string) {
+        if (existsSync(gitSecretsExecutablePath)) {
+            console.log('INFO: git-secrets already installed')
+            return
+        }
+        console.log('INFO: Installing git-secrets...')
+        spawnSync('mkdir', ['-p', path.parse(gitSecretsExecutablePath).dir])
+        spawnSync('curl', [
+            '-o',
+            gitSecretsExecutablePath,
+            'https://raw.githubusercontent.com/awslabs/git-secrets/99d01d58ebcc06e237c0e3f3ff5ae628aeef6aa6/git-secrets',
+        ])
+        spawnSync('chmod', ['+x', gitSecretsExecutablePath])
+    }
+
+    /** git-secrets patterns that will not cause a failure during the scan */
+    function setAllowListPatterns() {
+        const allowListPatterns: string[] = ['^node_modules/', '^\\.[a-zA-Z]', '^dist/']
+
+        allowListPatterns.forEach(pattern => {
+            spawnSync(gitSecrets, ['--add', '--allowed', pattern], { cwd: toolkitProjectDir })
+        })
+    }
+
+    function setDenyListPatterns() {
+        const denyListPatterns: string[] = []
+
+        denyListPatterns.forEach(pattern => {
+            spawnSync(gitSecrets, ['--add', pattern], { cwd: toolkitProjectDir })
+        })
+    }
+
+    before(function () {
+        if (platform() === 'win32') {
+            this.skip()
+        }
+
+        toolkitProjectDir = path.join(getProjectDir(), '..', '..')
+        testFixturesPath = path.join(toolkitProjectDir, 'src', 'testFixtures', 'resources')
+        mkdirSync(testFixturesPath)
+        accessKeyFilePath = path.join(testFixturesPath, 'fileWithAccessKey.ts')
+        gitSecrets = path.join(testFixturesPath, 'git-secrets')
+
+        setupGitSecretsExecutable(gitSecrets)
+
+        // Register all patterns with `git-secrets`
+        spawnSync(gitSecrets, ['--register-aws'], { cwd: toolkitProjectDir })
+        setDenyListPatterns()
+        setAllowListPatterns()
+    })
+
+    afterEach(function () {
+        if (existsSync(accessKeyFilePath)) {
+            unlinkSync(accessKeyFilePath)
+        }
+    })
+
+    it('ensures no git secrets are found', function () {
+        const result = spawnSync(gitSecrets, ['--scan', '--no-index'], { cwd: toolkitProjectDir })
+        assert.strictEqual(result.status, 0, `Failure output: ${result.stderr.toString()}`)
+    })
+
+    it('sanity check it finds secrets', function () {
+        // Create file in project that has secret key value.
+        // Need to build access key string incrementally to not trigger git-secrets.
+        const keyValue = 'yAki21XLhAIBiKvyaxr4p/ltr8OxkZTHISISFAKE'
+        const mySecretAccessKey = `const x = { "aws_secret_access_key": "${keyValue}" }`
+        const fileContent = dedent(
```

Comment:
This is nice to have, but is it worth an extra dependency? looks like it has no transitive dependencies, but still not sure.

Reply:
Not a necessity to me, will remove